Library/Homebrew/os/linux/global.rb

```diff
@@ -1,7 +1,12 @@
 # frozen_string_literal: true
 
 # enables experimental patchelf.rb write support.
-HOMEBREW_PATCHELF_RB_WRITE = ENV["HOMEBREW_PATCHELF_RB_WRITE"].present?.freeze
+HOMEBREW_PATCHELF_RB_WRITE = (
+  (ENV["HOMEBREW_PATCHELF_RB_WRITE"].present? ||
+  ENV["HOMEBREW_DEVELOPER"].present?) &&
+  !ENV["CI"].present? &&
+  !ENV["NO_HOMEBREW_PATCHELF_RB_WRITE"].present?
```

Comment:
```suggestion
  !ENV["HOMEBREW_NO_PATCHELF_RB_WRITE"].present? &&
  (ENV["HOMEBREW_PATCHELF_RB_WRITE"].present? ||
    (!ENV["CI"].present? && ENV["HOMEBREW_DEVELOPER"].present?))
```
if `HOMEBREW_NO_PATCHELF_RB_WRITE` then *off*
if `HOMEBREW_PATCHELF_RB_WRITE` then *on*
if `CI` then *off*
if `HOMEBREW_DEVELOPER` then *on*
otherwise *off*